### PR TITLE
Add transformer

### PR DESCRIPTION
**Pull request by jeffrey-hsu**
* add transformer/ folder with all model modules
* add transformer based seq2seq english to french translations
* add tensor board visualizations to embeddings notebook

### CODE REVIEW

File: 2_embeddings.ipynb [hunk 1]
```diff
@@ -16,29 +16,15 @@
   },
```
Comment (suzil) [inline 2_embeddings.ipynb:16]: <p>Command grouping here seems a bit arbitrary but no big deal</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/scoutbee/pytorch-nlp-notebooks/pull/19/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (suzil) [inline 2_embeddings.ipynb:16]: <p><code>TensorBoardColab('runs/')</code>, don't need to assign it to a value I'm pretty sure, would be nice to call the dir <code>runs/</code> or something as that's more normal too</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/scoutbee/pytorch-nlp-notebooks/pull/19/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='26'/>

File: 6_transformer_translation.ipynb [hunk 2]
```diff
@@ -0,0 +1,1266 @@
+{
```
Comment (suzil) [inline 6_transformer_translation.ipynb:1]: <p><code>.9999</code> how many lines of data are there??</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/scoutbee/pytorch-nlp-notebooks/pull/19/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

Comment (suzil) [inline 6_transformer_translation.ipynb:1]: <p><code>print(f'Total number of trainable parameters: {total_params:,}')</code></p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/scoutbee/pytorch-nlp-notebooks/pull/19/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='17'/>

Comment (suzil) [inline 6_transformer_translation.ipynb:1]: <p><code>tgt</code> seems more target than <code>trg</code> to me, but doesn't matter</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/scoutbee/pytorch-nlp-notebooks/pull/19/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='19'/>

**Review by suzil — APPROVED** [review]

File: 2_embeddings.ipynb (hunk 1 above)
Comment (jeffrey-hsu) [reply 2_embeddings.ipynb:16]: I got an error on colab when running this line, but without it TensorBoard still runs, so I commented out.

File: 6_transformer_translation.ipynb (hunk 2 above)
Comment (jeffrey-hsu) [reply 6_transformer_translation.ipynb:1]: 145437 in total, I just want a few lines of prediction output in test_loader

**Pull request merged**